### PR TITLE
Resolves issue #25 - Improper handling of quoted arguments with white space

### PR DESCRIPTION
**Pull request by ckstettler**
Resolves issue #14 and Resolves issue #25 

- Added logic to normalize arguments by looking for white space and recreating the match arguments with those arguments with spaces into one quoted argument.
Added additional test cases and examples
- This change does have a compatibility change from previous implementation.

If a test case has input '"a b" c' today then previously the capture[0] = "a b c" would not be capture[0] = '"a b" c'.

Like wise if the old match logic was --match "a b c" and the stub was called with "a b" c then the match logic will no longer work with the new approach.

### CODE REVIEW

**Review by ckstettler — COMMENTED** [review]
lgtm

**Comment by ckstettler** [comment]
@CaptainQuirk  I am not able to merge my own PR.  Since you have tested this one can you review the PR and either merge or provide comments?

File: bin/shellmock
Comment (CaptainQuirk) [inline bin/shellmock:65]: Is there a reason why you're using the backtick notation for command substitution rather than `$(..)` ? I've seen both notations throughout the code and was wondering what was the rationale to use one over the other

**Review by CaptainQuirk — CHANGES_REQUESTED** [review]

File: README.adoc
Comment (ckstettler) [reply README.adoc:60]: Updated.

File: README.adoc
Comment (ckstettler) [reply README.adoc:62]: The idea was to enable debugging in conjunction with TEST_FUNCTION in bats.  My rationale was if you are trying to isolate and run a single test then you are likely debugging.  If you are not using skipIfNot in your bats file then I can see that tests would run and debug would happen.

If configured as described in the read me with skipIfNot then I would expect if you set TEST_FUNCTION to foo then no tests would run.

I could make them mutually exclusive.  It is this way primarily because this is how I was doing test debugging. 

If there is an need to separate them lets raise a separate issue and I can take a look, or you could propose something in a PR if you like.

File: README.adoc
Comment (ckstettler) [reply README.adoc:295]: Yes your understanding is correct.   That is because when the users script calls the mock the quotes are interpreted and eaten by the shell.
- grep "a b" bob.c
- grep 'a b' bob.c

The two above do the exact same thing and from inside the grep stub I can tell which was passed because there are no quotes at all any more.


As far as the documentation, that is a fair ask.  I was thinking also that the readme should maybe correlate with the samples.bats.  I could spend more time describing each in more detail.

File: bin/shellmock
Comment (ckstettler) [reply bin/shellmock:65]: I can certainly see that I was inconsistent.  I typically use the ` because that is what I had learned first I think.  However, I can not explain the inconsistent usage.

I can update the code to prefer $().

File: README.adoc
Comment (CaptainQuirk) [reply README.adoc:62]: No that's fine by me ! I'm actually satisfied with the way it works and it seems I have misread the documentation

File: bin/shellmock
Comment (CaptainQuirk) [reply bin/shellmock:65]: I've been using [shellcheck](https://github.com/koalaman/shellcheck) to lint my bash code in the last months. Would it be something worth adding ?

Comment (ckstettler) [reply bin/shellmock:65]: I will look into that.

**Comment by ckstettler** [comment]
I am also working on getting you the access to do the merge.  Hopefully that will happen today.

I will make two updates prior to the merge:
- Fix the $() for consistency
- Update the readme with current samples.bats

**Comment by ckstettler** [comment]
The access should have been granted.  Please confirm whether you have merge ability please.  Thank you so much.

**Comment by ckstettler** [comment]
Okay.  I made quite a few documentation updates including some additional examples.  I have plans to add more samples to cover all of the support features.

I will probably submit another PR and add shellcheck fixes.  I think this one is good enough.  If you have time @CaptainQuirk for a second review that would be great.

**Review by CaptainQuirk — APPROVED** [review]
Looks good to me !

Sorry for the massive delay !

**Comment by ckstettler** [comment]
You should have gotten an invite to be an outside collaborator.  That would give you the rights to push the merge button. Did you see it?

**Comment by CaptainQuirk** [comment]
Well I don't ;-(

**Comment by ckstettler** [comment]
A second one was sent today, but you did not see it either I am guessing?

**Comment by CaptainQuirk** [comment]
Oh no I joined the organisation all right, but I don't see any button to merge

**Comment by CaptainQuirk** [comment]
Capture du 2020-10-14 22-57-09

**Comment by tmbjmu** [comment]
@ckstettler @CaptainQuirk Please try now. I've added you to the write team and codeowners file

**Comment by CaptainQuirk** [comment]
There need to be another review apparently : I see the button but it is disabled

**Review by sheosinha — APPROVED** [review]

**Review by tmbjmu — APPROVED** [review]

**Pull request merged**
